packages/l10n/src/types.ts

```diff
@@ -33,7 +33,7 @@ export type FieldNameTranformationMapping = {
 
 export type FormatLocalizedStringOptions<T> = {
   key: keyof T;
-  locale: string;
+  locale: string | null;
```

Comment:
I want to allow `null`  here.
Normally, the source of `dataLocale` is extracted from `application-context` and that value allows `null` 

[reference](https://github.com/commercetools/merchant-center-application-kit/blob/master/packages/application-shell-connectors/src/components/application-context/application-context.tsx#L117:L117)

Reply:
This is the case for applications that do not have a project context, for example auth and account.

I think we can be strict and avoid `null` here.

Reply:
After a second thought, it would be annoying indeed to always let the consumer check for the existence of `dataLocale`.
I suppose we would need to fix the types there first, but it's tricky to work with conditionals.